src/ui/components/Timeline/index.tsx

```diff
@@ -65,8 +67,15 @@ export default function Timeline() {
       return;
     }
 
-    dispatch(setTimelineToTime(mouseTime, true));
-    dispatch(setTimelineState({ currentTime: mouseTime }));
+    const paintOrMouseEvent = mostRecentPaintOrMouseEvent(mouseTime);
+    if (paintOrMouseEvent?.point) {
+      if (!dispatch(seek(paintOrMouseEvent.point, mouseTime, false))) {
+        // if seeking to the new point failed because it is in an unloaded region,
+        // we reset the timeline back to the current time
+        setTimelineToTime(ThreadFront.currentTime);
+        setTimelineState({ currentTime: ThreadFront.currentTime });
```

Comment:
I wish we could wrap up all of this logic (time and possibly point) to a single action creator so that views didn't have to know to call all 3.